tower/tests/spawn_ready/main.rs

```diff
@@ -43,3 +43,22 @@ async fn when_inner_fails() {
         "foobar"
     );
 }
+
+#[tokio::test(flavor = "current_thread")]
+async fn propagates_trace_spans() {
+    use tower::{util::ServiceExt, Service};
+    use tracing::Instrument;
+
+    time::pause();
+    let _t = support::trace_init();
+
+    let span = tracing::info_span!("my_span");
+
+    let service = support::AssertSpanSvc::new(span.clone());
+    let mut service = SpawnReady::new(service);
+    let result =
+        tokio::spawn(async move { service.ready_and().await?.call(()).await }.instrument(span));
+
+    time::sleep(time::Duration::from_millis(100)).await;
```

Comment:
why is the sleep necessary? perhaps a comment explaining it?

Reply:
If we really do need a sleep, can we use `time::pause`?

Reply:
there's a `time::pause` here: https://github.com/tower-rs/tower/blob/0408e315b5eef267e070d9aa37dbae900c1676b0/tower/tests/spawn_ready/main.rs#L52

I added this because the other `SpawnReady` tests have a sleep...although after looking at it more closely I don't think it's necessary in any case. I'll remove them in a separate branch.

Reply:
to elaborate: i'm _guessing_ the sleep was originally added to give the background task time to run, and then someone did a pass where they came through and slapped a `time::pause` in every test that has a sleep in it...so now, i think the sleep basically just does nothing.

Reply:
ah, yeah, that's me

Reply:
lol...i think a more robust solution would be just changing these tests to use `ready_and().await` instead of sleeping?

Reply:
i could see sleeping where we poll once (as i imagine this was initially), but with await it seems totally unnecessary.